### ISSUE TITLE
consider re-organizing S3 class hierarchy for tfestimator objects

### ISSUE THREAD

**Issue by kevinushey** [rstudio/tfestimators#53]
Consider the class currently applied for e.g. the output of `linear_regressor()`:

```
> class(tf_model)
[1] "tf_model"  "linear"    "regressor"
```

I think there's a couple tweaks we could make here:

1. Should these have class `tf_estimator` rather than `tf_model`? The underlying object really is just an R object wrapping over an estimator, so it is perhaps more accurate to call it that.

2. We should also prefix the extra classes with `tf_estimator` (or `tf_model`) -- standalone S3 class names e.g. `linear` feel a bit odd to me.

E.g. for `linear_regressor()` I think the class output should be more like:

```
> class(tf_model)
[1] "tf_estimator"  "tf_estimator_regressor"    "tf_estimator_regressor_linear"
```

In other words, I think the three main things worth encoding in the class of the object are:

1. It's an estimator,
2. It's a regressor vs. it's a classifier,
3. The actual full class of the estimator.

@jjallaire @terrytangyuan, let me know if you think this sounds sensible, or if there are any other bits worth adding as a class. (For example, do you think it's necessary that we differentiate between 'canned' estimators provided by the `tfestimators` package, and user-defined estimators? It would be nice if there were effectively no difference and we didn't have to customize behaviors based on whether we defined the estimator vs. the user defined the estimator)
